src/server/main.cpp

```diff
@@ -53,61 +115,26 @@ void _ZN6QImageC1EiiNS_6FormatE(void *that, int x, int y, int f) {
   qImageCtor(that, x, y, f);
 }
 
-void doUpdate(const SwtFB &fb, const swtfb_update &s) {
-  auto mxcfb_update = s.update;
-  auto rect = mxcfb_update.update_region;
-#ifdef DEBUG_DIRTY
-  std::cerr << "Dirty Region: " << rect.left << " " << rect.top << " "
-            << rect.width << " " << rect.height << endl;
-#endif
-
-  int waveform = mxcfb_update.waveform_mode;
-  // full = 1, partial = 0
-  auto update_mode = mxcfb_update.update_mode;
-
-  // For now, we are using two waveform modes:
-  // 1: DU - direct update, fast
-  // 2: GC16 - high fidelity (slow)
-  // 3: GL16 - what the rm is using
-  if (waveform > 3) {
-    waveform = 3;
-  }
-  auto flags = 0;
-
-  if (waveform == 1 && update_mode == 0) {
-    // fast draw
-    flags = 4;
-  } else if (update_mode == 1 && waveform == 2) {
-    // blink
-    flags = 3;
-  }
-#ifdef DEBUG
-  cerr << "doUpdate " << endl;
-  cerr << "mxc: waveform_mode " << mxcfb_update.waveform_mode << endl;
-  cerr << "mxc: update mode " << mxcfb_update.update_mode << endl;
-  cerr << "mxc: update marker " << mxcfb_update.update_marker << endl;
-  cerr << "final: waveform " << waveform;
-  cerr << " flags " << flags << endl << endl;
-#endif
-  if (mxcfb_update.update_marker > 0) {
-    fb.WaitForLastUpdate();
-  }
```

Comment:
why was this deleted? since this is deleted, likely need to audit a few apps to make sure they still work, specifically KOReader. 

there is some strangeness in general, since the API for waiting for update in mxcfb is 1) send an update with an update_marker and then 2) use a special ioctl (MXCFB_WAIT_FOR_UPDATE or something similar) with that marker, which then waits for that update to finish. 

with our current API in rM2, we have 'waitForLastUpdate()', but it doesn't take an update marker. therefore, we assume that if a program uses update_marker, we will automatically call waitForLastUpdate() on their behalf, even though this is not the correct behavior.

Reply:
although, now i see the wait() call is before the draw(), which definitely seems wrong

Reply:
Because it's a no-op, in the static method call selector from qt:
 ```
case 5: // idx 5 == WaitForLastUpdate
  return
```

The new method for waiting is setting the `sync` bit in do update (bit 2).